packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -240,6 +240,17 @@ export class Secret extends SecretBase {
     });
 
     this.encryptionKey = props.encryptionKey;
+
+    // @see https://docs.aws.amazon.com/kms/latest/developerguide/services-secrets-manager.html#asm-authz
+    this.encryptionKey?.grant(
+      new kms.ViaServicePrincipal(`secretsmanager.${Stack.of(this).region}.amazonaws.com`, new iam.AccountPrincipal(Stack.of(this).account)),
```

Comment:
I wonder... would instead calling

```ts
this.encryptionKey?.grantEncryptDecrypt(new iam.ViaServicePrincipal(...));
```

be possible here? I think KMS is the better place to keep this list of actions in.

Reply:
Looks like `grantEncryptDecrypt` is missing a couple of the required permissions, but I can split this up. Let kms handle the encrypt/decrypt permissions, and then add in the extras using a normal grant.

Reply:
I made this change. The trade off was is increased size of the resulting Cfn template since the `iam.Grant` functionality doesn't try to common policy statements that can be merged.